modules/react-components/src/grid/template-grid.tsx

```diff
@@ -0,0 +1,273 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ImageUtils, URLUtils } from "@wso2is/core/utils";
+import _ from "lodash";
+import React, { FunctionComponent, ReactElement, ReactNode, SyntheticEvent, useEffect, useState } from "react";
+import { Grid } from "semantic-ui-react";
+import { LinkButton } from "../button";
+import { TemplateCard, TemplateCardPropsInterface } from "../card";
+import { Heading } from "../typography";
+
+/**
+ * Proptypes for the template grid component.
+ */
+interface TemplateGridPropsInterface {
+    /**
+     * Empty placeholder
+     */
+    emptyPlaceholder?: ReactElement;
+    /**
+     * Heading for the grid.
+     */
+    heading?: ReactNode;
+    /**
+     * Callback to be fired on template selection.
+     */
+    onTemplateSelect: (e: SyntheticEvent, { id }: { id: string }) => void;
+    /**
+     * Enable/ Disable pagination.
+     */
+    paginate?: boolean;
+    /**
+     * Pagination limit.
+     */
+    paginationLimit?: number;
+    /**
+     * Grid pagination options.
+     */
+    paginationOptions?: TemplateGridPaginationOptionsInterface;
+    /**
+     * Sub heading for the grid.
+     */
+    subHeading?: ReactNode;
+    /**
+     * Title for the tags section.
+     */
+    tagsSectionTitle?: TemplateCardPropsInterface["tagsSectionTitle"];
+    /**
+     * List of templates.
+     */
+    templates: any;
+    /**
+     * Template icons.
+     */
+    templateIcons?: object;
+    /**
+     * Grid type.
+     */
+    type: "application" | "idp";
+}
+
+/**
+ * Interface for grid pagination options.
+ */
+export interface TemplateGridPaginationOptionsInterface {
+    /**
+     * Show more button label.
+     */
+    showMoreButtonLabel: string;
+    /**
+     * Show less button label.
+     */
+    showLessButtonLabel: string;
+}
+
+
+/**
+ * Initial display limit.
+ * TODO: Generate limit dynamically with screen dimensions.
+ * @type {number}
+ */
+const DEFAULT_PAGINATION_LIMIT = 5;
+
+/**
+ * Template grid component.
+ *
+ * @param {TemplateGridPropsInterface} props - Props injected to the component.
+ *
+ * @return {React.ReactElement}
+ */
+export const TemplateGrid: FunctionComponent<TemplateGridPropsInterface> = (
+    props: TemplateGridPropsInterface
+): ReactElement => {
+
+    const {
+        emptyPlaceholder,
+        heading,
+        onTemplateSelect,
+        paginate,
+        paginationLimit,
+        paginationOptions,
+        subHeading,
+        tagsSectionTitle,
+        templates,
+        templateIcons,
+        type
+    } = props;
+
+    const [ templateList, setTemplateList ] = useState<any>([]);
```

Comment:
Shall we add a proper type for the state variable?

Reply:
Fixed with commit `2ab4dc8`